### PR TITLE
Fix for Angular 9

### LINKED ISSUE #21: Error after upgrade to Angular 9
Hello,

After upgrading to Angular 9, I'm getting the following error;

> ERROR in The target entry-point “ngx-powerbi” has missing dependencies: - service

### PR DESCRIPTION
**Pull request by beradrian**
Fixes #21

### CODE REVIEW

**Review by kamilslusarczyk — CHANGES_REQUESTED** [review]
Hello, 

I have followed your changes from below PR to check if this works for app, I develop. Everything went cool apart from one thing - ɵɵFactoryDef in both component and service throws an error, that it requires more generic arguments. I am not sure, if I have done something incorrectly with your changes, but update to Angular v 9.1.1 fixed my problem (below Stack Overflow's thread was an inspiration).

https://stackoverflow.com/questions/60898133/angular-9-ivy-generic-type-%C9%B5%C9%B5factorydef-requires-2-type-arguments

**Comment by beradrian** [comment]
What you're suggesting is an upgrade to 9.1.1? Your app could use any Angular 9.x version anyway.

File: projects/ngx-powerbi/package.json [hunk 1]
```diff
@@ -24,8 +27,9 @@
     "reporting"
   ],
   "peerDependencies": {
-    "@angular/common": "^7.0.0",
-    "@angular/core": "^7.0.0",
-    "powerbi-client": "^2.6.5"
+    "@angular/common": "^9.0.6",
+    "@angular/core": "^9.0.6",
```
Comment (ramandeep-singh-1983) [inline projects/ngx-powerbi/package.json]: Angular versions in main package.json are ^9.0.1 but here the versions are ^9.0.6, can you make them same? Better option would be to update angular versions to the latest one i.e. 9.1.2

File: package.json [hunk 2]
```diff
@@ -10,46 +10,48 @@
...
-    "core-js": "^2.5.4",
-    "ng-packagr": "^4.4.0",
+    "@angular/animations": "~9.0.1",
+    "@angular/common": "~9.0.1",
+    "@angular/compiler": "~9.0.1",
+    "@angular/core": "~9.0.1",
+    "@angular/forms": "~9.0.1",
+    "@angular/platform-browser": "~9.0.1",
+    "@angular/platform-browser-dynamic": "~9.0.1",
+    "@angular/router": "~9.0.1",
```
Comment (ramandeep-singh-1983) [inline package.json]: can you update these to the latest version i.e. 9.1.2?

File: package.json [hunk 3]
```diff
@@ -10,46 +10,48 @@
...
-    "@angular-devkit/build-angular": "~0.10.0",
-    "@angular/cli": "~7.0.5",
-    "@angular/compiler-cli": "^7.0.0",
-    "@angular/language-service": "^7.0.0",
-    "@types/jasmine": "~2.8.8",
+    "@angular-devkit/build-angular": "~0.900.2",
+    "@angular-devkit/build-ng-packagr": "~0.900.6",
+    "@angular/cli": "~9.0.2",
+    "@angular/compiler-cli": "~9.0.1",
+    "@angular/language-service": "~9.0.1",
```
Comment (ramandeep-singh-1983) [inline package.json]: can you update these to the latest version i.e. 9.1.2?

File: projects/ngx-powerbi/src/lib/ngx-powerbi.component.ts [hunk 4]
```diff
@@ -41,7 +41,7 @@ export class NgxPowerBiComponent implements OnInit, OnChanges, OnDestroy {
   @Input() name: string;
   @Input() options: models.ISettings;
   @Output() embedded = new EventEmitter<number>();
-  @ViewChild('ngxPowerBiIFrame') ngxPowerBiIFrameRef: ElementRef;
+  @ViewChild('ngxPowerBiIFrame', {static: true}) ngxPowerBiIFrameRef: ElementRef;
```
Comment (ramandeep-singh-1983) [inline projects/ngx-powerbi/src/lib/ngx-powerbi.component.ts:44]: I hope you have tested the OnInit -> AfterViewInit and this {static: true} change?

File: projects/ngx-powerbi/tsconfig.lib.json [hunk 5]
```diff
@@ -2,27 +2,15 @@
...
     "declaration": true,
-    "sourceMap": true,
     "inlineSources": true,
-    "emitDecoratorMetadata": true,
-    "experimentalDecorators": true,
-    "importHelpers": true,
-    "types": [],
-    "lib": [
-      "dom",
-      "es2015"
```
Comment (ramandeep-singh-1983) [inline projects/ngx-powerbi/tsconfig.lib.json:17]: Reason for removal of these lines? Was this done automatically on upgrade to Angular 9.x?

**Review by ramandeep-singh-1983 — CHANGES_REQUESTED** [review]
Please see my comments in the PR.

File: projects/ngx-powerbi/package.json [hunk 6]
```diff
@@ -24,8 +27,9 @@
...
+    "powerbi-client": "^2.6.5",
```
Comment (ramandeep-singh-1983) [inline projects/ngx-powerbi/package.json]: Maybe it makes sense to update this to 2.11.0

File: package.json [hunk 7]
```diff
@@ -10,46 +10,48 @@
...
-    "ng-packagr": "^4.4.0",
+    "@angular/animations": "~9.0.1",
+    "@angular/common": "~9.0.1",
+    "@angular/compiler": "~9.0.1",
+    "@angular/core": "~9.0.1",
+    "@angular/forms": "~9.0.1",
+    "@angular/platform-browser": "~9.0.1",
+    "@angular/platform-browser-dynamic": "~9.0.1",
+    "@angular/router": "~9.0.1",
     "powerbi-client": "^2.6.5",
```
Comment (ramandeep-singh-1983) [inline package.json]: Maybe, it makes sense to update this to 2.11.0

File: projects/ngx-powerbi/src/lib/ngx-powerbi.component.ts (hunk 4 above)
Comment (beradrian) [reply projects/ngx-powerbi/src/lib/ngx-powerbi.component.ts:44]: I was already using ngx-powerbi in my application and I had to do this change to make it work with Angular 9.

File: projects/ngx-powerbi/tsconfig.lib.json (hunk 5 above)
Comment (beradrian) [reply projects/ngx-powerbi/tsconfig.lib.json:17]: I think this was done during the upgrade.

Comment (beradrian) [reply projects/ngx-powerbi/tsconfig.lib.json:17]: I can review them and added them back again if you want.

File: package.json (hunk 3 above)
Comment (beradrian) [reply package.json]: I'll do.

File: projects/ngx-powerbi/package.json (hunk 6 above)
Comment (beradrian) [reply projects/ngx-powerbi/package.json]: I'll look into it.

File: projects/ngx-powerbi/tsconfig.lib.json (hunk 5 above)
Comment (ramandeep-singh-1983) [reply projects/ngx-powerbi/tsconfig.lib.json:17]: it's ok, as long as it works

File: projects/ngx-powerbi/package.json (hunk 1 above)
Comment (beradrian) [reply projects/ngx-powerbi/package.json]: I updated to 9.1.2

File: package.json (hunk 2 above)
Comment (beradrian) [reply package.json]: Done.

File: package.json (hunk 3 above)
Comment (beradrian) [reply package.json]: Done.

File: projects/ngx-powerbi/package.json (hunk 6 above)
Comment (beradrian) [reply projects/ngx-powerbi/package.json]: I'm using in my project 2.11.0 and it works fine. I updated it.

File: package.json (hunk 7 above)
Comment (beradrian) [reply package.json]: I'm using in my project 2.11.0 and it works. I updated it.

**Comment by beradrian** [comment]
> 
> 
> Please see my comments in the PR.

I responded to your comments and updated to latest version of Angular and PowerBI client.

**Review by ramandeep-singh-1983 — APPROVED** [review]
Changes look good

**Comment by ramandeep-singh-1983** [comment]
@beradrian, i have approved the PR. Just want to confirm if the code is fully tested with Angular 9.1.2, so please let me know.

Question: is the squash and merge button enabled for you now or do i need to merge it?
I will probably change the release version to 3.0.0 as this looks like a major update as we will be supporting Angular 9+

**Comment by beradrian** [comment]
@ramandeep-singh-1983 I'm using this library in my project and tested it with Angular 9.
I'm not able to merge it myself. Only those with write access to this repository can merge pull requests.

**Pull request merged**